controllers/release/release_adapter.go

```diff
@@ -345,6 +356,12 @@ func (a *Adapter) registerGitOpsDeploymentStatus(binding *applicationapiv1alpha1
 		return nil
 	}
 
+	// Do nothing if the release does not own the binding
+	if binding.GetAnnotations()[libhandler.TypeAnnotation] != a.release.GetObjectKind().GroupVersionKind().GroupKind().String() ||
```

Comment:
This check should be in the operation itself and call `registerGitOpsDeploymentStatus` with a valid binding only. First time I went over the operation made me wonder if you were taking into account the ownership.

Reply:
fixed